### ISSUE TITLE
Focus Zone sets tabIndex to 0 for unfocusable element when it updates

### ISSUE THREAD

**Issue by siddharthkp** [primer/behaviors#138]
- via https://github.com/primer/react/issues/2270

**Describe the bug**
If an element is given a tabIndex of -1 and filtered out of a Focus Zone with focusableElementFilter, it should not participate in the Focus Zone or be tabbable.  This works as expected until some prop on the unfocusable element changes. When that happens, the Focus Zone will update the tabIndex of that element to be 0, making it tabbable which not desired.  This seems like an edge case, but it is occurring in the ReposFileTree component, which contains expand/collapse buttons in each row that should not participate in the Focus Zone.

**Note:** As this is a public repo, please be careful not to include details or screenshots from unreleased GitHub products or features. In most cases, a good bug report should be able to describe the new component without including business logic or feature details, but if you must discuss context relating to an unreleased feature, please open an issue in the private [Design Systems repo](https://github.com/github/design-systems/issues/new/choose) and link to it here.

**To Reproduce**
Here's a simple example:
```
export const DefaultButton = (args: ButtonProps) => {
  const [expanded, setExpanded] = useState(false)
  const {containerRef} = useFocusZone({
    focusableElementFilter: element => element.getAttribute('data-focusable') !== 'false'
  })
  return (
    <Box ref={containerRef as RefObject<HTMLDivElement>}>
      <Button {...args}>Default</Button>
      <Button {...args}>Default2</Button>
      <Button
        {...args}
        data-focusable="false"
        tabIndex={-1}
        aria-expanded={expanded}
        onClick={() => {
          setExpanded(!expanded)
        }}
      >
        Default3
      </Button>
      <Button {...args}>Default4</Button>
    </Box>
  )
}
```
1. Paste into the DefaultButton storyboook example in this repo
2. Inspect the Default3 Button and notice it has a tabIndex of -1
3. Click on the Default3 Button and notice its tabIndex changes to 0 

**Expected behavior**
Elements that are being filtered out of the focus zone with focusableElementFilter should not have their tabIndex updated by it.

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser chrome
 - Version 103.0.5060.134 (Official Build) (x86_64)

**Additional context**

**Comment by siddharthkp** [comment]
The problem seems to be in the `focusin` handler where it simply selects the interacted element without checking if it's part of `focusableElements` if the focusInStrategy is `previous` (which is the default) 

https://github.com/primer/behaviors/blob/main/src/focus-zone.ts#L568-L569

**Comment by siddharthkp** [comment]
Still valid
